### ISSUE TITLE
C++20 header `version` suggested for `std::ptrdiff_t`

### ISSUE THREAD

**Issue by firewave** [include-what-you-use/include-what-you-use#1478]
```
/__w/cppcheck/cppcheck/externals/simplecpp/simplecpp.cpp should add these lines:
#include <version>        // for std::ptrdiff_t

/__w/cppcheck/cppcheck/externals/simplecpp/simplecpp.cpp should remove these lines:
- #include <cstddef>  // lines 17-17
```

`version` is a C++20 header: https://en.cppreference.com/w/cpp/header/version. Our code specifies `-std=c++11`.
According to https://en.cppreference.com/w/cpp/types/ptrdiff_t the symbol is provided by `cstddef`.

This started with (switching to) IWYU 0.21 from OpenSUSE. We previously used 0.20 from Fedora. So it might also be related to the distro.

As I have no local system with that version I do not have sample code at the moment.

**Comment by kimgr** [comment]
I think you may be getting bitten by this mapping: https://github.com/include-what-you-use/include-what-you-use/blob/master/iwyu_include_picker.cc#L865

Both `std::size_t` and `std::ptrdiff_t` are defined in `bits/c++config.h`, which is in itself a little weird, but I see no reason for us to map that basic header to any public headers.

**Comment by kimgr** [comment]
I was just about to throw in a fix for this, but I can no longer reproduce... That's confusing. Here's my repro experiment:
```
#include <cctype>  // includes bits/c++config.h, which has std::ptrdiff_t
#include <cstddef>

void foo() {
  std::ptrdiff_t x = (char*)100 - (char*)0;
}
```

This responds as I would expect:
```
$ include-what-you-use tests/ptrdiff/t.cc 

tests/ptrdiff/t.cc should add these lines:

tests/ptrdiff/t.cc should remove these lines:
- #include <cctype>  // lines 1-1

The full include-list for tests/ptrdiff/t.cc:
#include <cstddef>  // for ptrdiff_t
---
```

This is on a system with gcc 11.4 and the corresponding libstdc++ 11 (assuming they're packaged together).

**Comment by kimgr** [comment]
I'll clean up the mappings here anyway, but it would be nice to have a clean repro to see that things behave as expected.

**Comment by kimgr** [comment]
@firewave do you have any way of testing: https://github.com/include-what-you-use/include-what-you-use/pull/1496?

**Comment by firewave** [comment]
> @firewave do you have any way of testing: #1496?

As it requires source changes I don't. I do not even have the version to reproduce it locally (yet).

**Comment by kimgr** [comment]
Fixed on master as of 5e104d6384c48242ac1c24037461896de64f192a.
